WireDomain/Sources/WireDomain/Helpers/AsyncSequence+ToStream.swift

```
@@ -0,0 +1,30 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+
+extension AsyncSequence {
+
+    func toStream() -> AsyncStream<Element> {
+        var iterator = makeAsyncIterator()
+        return AsyncStream {
+            try? await iterator.next()
```

hard to see the impact, but the error is swallowed here ?

---

It's because it's an `AsyncStream` so you can't throw. I'll make this an `AsyncThrowingStream` instead.

---

done 3eae8e6ecf4d04a0c301ed2ae1f1b892f12960ce